src/setup/rgw.py

```diff
@@ -573,31 +577,53 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf):
         """ Configure logrotate utility for rgw logs."""
-        log_dir = conf.get(LOG_PATH_KEY)
-        log_file_path = os.path.join(log_dir, COMPONENT_NAME, Rgw._machine_id)
+        log_dir = conf.get(const.LOG_PATH_KEY)
+        log_file_path = os.path.join(log_dir, const.COMPONENT_NAME, Rgw._machine_id)
         # create radosgw logrotate file.
         # For eg:
         # filepath='/etc/logrotate.d/radosgw'
-        old_file = os.path.join(LOGROTATE_DIR, 'ceph')
+        old_file = os.path.join(const.LOGROTATE_DIR, 'ceph')
         if os.path.exists(old_file):
             os.remove(old_file)
         try:
-            with open(LOGROTATE_TMPL, 'r') as f:
+            with open(const.LOGROTATE_TMPL, 'r') as f:
                 content = f.read()
             content = content.replace('TMP_LOG_PATH', log_file_path)
-            with open(LOGROTATE_CONF, 'w') as f:
+            with open(const.LOGROTATE_CONF, 'w') as f:
                 f.write(content)
             Log.info(f'{LOGROTATE_TMPL} file copied to {LOGROTATE_CONF}')
         except Exception as e:
-            Log.error(f"Failed to configure logrotate for {COMPONENT_NAME}. ERROR:{e}")
+            Log.error(f"Failed to configure logrotate for {const.COMPONENT_NAME}. ERROR:{e}")
 
     @staticmethod
     def _verify_backend_store_value(conf: MappedConf):
         """Verify backed store value as motr."""
         config_file = Rgw._get_rgw_config_path(conf)
         Rgw._load_rgw_config(Rgw._rgw_conf_idx, f'ini://{config_file}')
         backend_store = Conf.get(Rgw._rgw_conf_idx, 'client>rgw backend store')
-        if not backend_store in SUPPORTED_BACKEND_STORES:
+        if not backend_store in const.SUPPORTED_BACKEND_STORES:
             raise SetupError(errno.EINVAL,
-                f'Supported rgw backend store are {SUPPORTED_BACKEND_STORES},'
+                f'Supported rgw backend store are {const.SUPPORTED_BACKEND_STORES},'
                 f' currently configured one is {backend_store}')
+
+    @staticmethod
+    def _update_rgw_config(conf: MappedConf, client_section: string, config_key_mapping: list):
+        """Update config properties from confstore to rgw config file."""
+        rgw_config_dir = Rgw._get_rgw_config_dir(conf)
+        rgw_config_file = os.path.join(rgw_config_dir, const.RGW_CONF_FILE)
+        Rgw._load_rgw_config(Rgw._rgw_conf_idx, f'ini://{rgw_config_file}')
+        Log.info(f'adding paramters to {client_section} in {rgw_config_file}')
+
+        # e.g config_key_mapping = [[confstore_key1, actual_rgw_config_key1],
+        # [confstore_key2, actual_rgw_config_key2], ..]
+        for confstore_key, config_key in config_key_mapping:
+            # fetch actual value of parameter from confstore.
+            config_value = Conf.get(Rgw._rgw_conf_idx, confstore_key)
+            if not config_value :
+                raise SetupError(errno.EINVAL,
+                    f'Confstore key/value is missing for key {confstore_key}')
```

Comment:
You can use '_get_cortx_conf' function to read key value from gconf.
config_value= Rgw._get_cortx_conf(conf, confstore_key)

Reply:
done